### DISCUSSION TITLE
Provided Google Play service account JSON has invalid format

### DISCUSSION THREAD

**Discussion by zingwena-softwares** [codemagic-ci-cd/codemagic-docs#2481] [Q&A] [answered]
### Use case

```Text
Im facing this error Provided Google Play service account JSON has invalid format

Please help
```

### Idea

```Text
I provided a base64 format on environment variables, so now I don't know which json format being referred to
```

**Comment by dtrdic** [comment, marked as answer]
Hi @zingwena-softwares please don't base64 the file, that's the invalid format for it. Just copy the content of the .json file and add it as an environment variable into a Codemagic.
Please check [this guide](https://docs.codemagic.io/yaml-publishing/google-play/#configure-publishing-in-codemagicyaml) as a reference

**Comment by zingwena-softwares** [comment]
Thank you @dtrdic , it worked


Now i'm facing this again 

*** What went wrong:
Execution failed for task ':app:signReleaseBundle'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.FinalizeBundleTask$BundleToolRunnable
   > Failed to read key talenthrms from store "/tmp/keystore.jks": DerInputStream.getLength(): lengthTag=111, too big.**

**Reply by dtrdic** [reply]
nice to hear that @zingwena-softwares, please make sure your android/build.gradle file looks like in [this guide](https://docs.codemagic.io/yaml-code-signing/signing-android/#signing-android-apps-using-gradle) in order to successfully signed your app.
Also please share a build id because i can't see what you are doing. Build id can be found on the build overview page

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
